cli/azd/internal/cmd/add/add_preview.go

```diff
@@ -0,0 +1,197 @@
+package add
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+	"log"
+	"strings"
+	"text/tabwriter"
+
+	"github.com/azure/azure-dev/cli/azd/pkg/infra/provisioning"
+	"github.com/azure/azure-dev/cli/azd/pkg/input"
+	"github.com/azure/azure-dev/cli/azd/pkg/output"
+	"github.com/azure/azure-dev/cli/azd/pkg/project"
+	"github.com/fatih/color"
+)
+
+// resourceMeta contains metadata of the resource
+type resourceMeta struct {
+	// The underlying resource type.
+	AzureResourceType string
+	// UseEnvVars is the list of environment variable names that would be populated when this resource is used
+	UseEnvVars []string
+}
+
+func metadata(r *project.ResourceConfig) resourceMeta {
+	res := resourceMeta{}
+
+	// These are currently duplicated, static values maintained separately from the backend generation files
+	// If updating resources.bicep, these values should also be updated.
+	switch r.Type {
+	case project.ResourceTypeHostContainerApp:
+		res.AzureResourceType = "Microsoft.App/containerApps"
+		res.UseEnvVars = []string{
+			strings.ToUpper(r.Name) + "_BASE_URL",
+		}
+	case project.ResourceTypeDbRedis:
+		res.AzureResourceType = "Microsoft.Cache/redis"
+		res.UseEnvVars = []string{
+			"REDIS_HOST",
+			"REDIS_PORT",
+			"REDIS_ENDPOINT",
+			"REDIS_PASSWORD",
+			"REDIS_URL",
+		}
+	case project.ResourceTypeDbPostgres:
+		res.AzureResourceType = "Microsoft.DBforPostgreSQL/flexibleServers/databases"
+		res.UseEnvVars = []string{
+			"POSTGRES_HOST",
+			"POSTGRES_USERNAME",
+			"POSTGRES_DATABASE",
+			"POSTGRES_PASSWORD",
+			"POSTGRES_PORT",
+			"POSTGRES_URL",
+		}
+	case project.ResourceTypeDbMongo:
+		res.AzureResourceType = "Microsoft.DocumentDB/databaseAccounts/mongodbDatabases"
+		res.UseEnvVars = []string{
+			"MONGODB_URL",
+		}
+	case project.ResourceTypeOpenAiModel:
+		res.AzureResourceType = "Microsoft.CognitiveAccounts/accounts/deployments"
+		res.UseEnvVars = []string{
+			"AZURE_OPENAI_ENDPOINT",
+			"Keyless (Microsoft Entra ID)",
```

Comment:
There's an environment variable floating around with this name?  Interesting!

Reply:
Removed this for now